src/Compilers/Core/Portable/SourceGeneration/SyntaxStore.cs

```diff
@@ -88,8 +88,23 @@ public IStateTable GetSyntaxInputTable(SyntaxInputNode syntaxInputNode, NodeStat
                             {
                                 try
                                 {
-                                    _cancellationToken.ThrowIfCancellationRequested();
-                                    syntaxInputBuilders[i].builder.VisitTree(root, state, model, _cancellationToken);
+                                    Stopwatch sw = Stopwatch.StartNew();
+                                    try
+                                    {
+                                        _cancellationToken.ThrowIfCancellationRequested();
+                                        syntaxInputBuilders[i].builder.VisitTree(root, state, model, _cancellationToken);
+                                    }
+                                    finally
+                                    {
+                                        var elapsed = sw.Elapsed;
+
+                                        // if this node isn't the one that caused the update, ensure we remember it and remove the time it took from the requester
+                                        if (syntaxInputBuilders[i].node != syntaxInputNode)
```

Comment:
>syntaxInputBuilders[i].node [](http://example.com/codeflow?start=44&length=27)

Consider extracting a local. (If the local is created at the start of the `for (...)` body, it can be used in the `catch` below as well.)